File: in_go/src/pattern_19.go
package src

import (
	"fmt"
)

func Pattern_19() {
	row := 4
	column := 7
	var symbol rune = 65
	halfColumn := column / 2

	for i := 0; i < row; i++ {
		symbol = 65
		for j := 0; j < column; j++ {
			if j >= halfColumn-i && j <= halfColumn+i {
				fmt.Print(string(symbol))
				if j < halfColumn {
					symbol++
				} else {
					symbol--
				}
			} else {
				fmt.Print(" ")
			}
		}
		fmt.Println()
	}
}

File: in_java/Pattern_19.java
package in_java;

public class Pattern_19 extends Pattern {
    public Pattern_19(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        byte symbol = (byte) this.symbol.charAt(0);
        int halfColumn = this.column / 2;
        for (int i = 0; i < this.row; i++) {
            symbol = (byte) this.symbol.charAt(0);
            for (int j = 0; j < this.column; j++) {
                if (j >= halfColumn - i && j <= halfColumn + i) {
                    System.out.print((char) symbol);
                    if (j < halfColumn) {
                        symbol++;
                    } else {
                        symbol--;
                    }
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}
